app/src/main/java/com/mapbox/maps/testapp/examples/linesandpolygons/MovingIconWithTrailingLineActivity.kt

```diff
@@ -239,6 +241,12 @@ class MovingIconWithTrailingLineActivity : AppCompatActivity() {
     )
   }
 
+  override fun onDestroy() {
+    super.onDestroy()
+    directionsClient?.cancelCall()
+    currentAnimator.cancel()
```

Comment:
```suggestion
        if(::currentAnimator.isInitialized){
      currentAnimator.cancel()
    }
```

Just make sure lateinit var is not null

Reply:
Let's make both `currentAnimator` and `directionsClient` use either `late init var` or `nullable var`